### PR TITLE
Ehn/pytest

### PR DESCRIPTION
**Pull request by mts299**
# Scope 

This PR focuses on creating a test suite that tests various parameter inputs to see if exceptions are raised when they shouldn't and should. Some functions will be tested for returns as well. 

This does not cover the appearance of plots, this will be the users'/testers job to ensure it looks correctly. 

**issue:** #179 

## Approval

**Number of approvals:** *2* but I will be merging other PRs into this one but keeping them small to make testing quicker 

## Test

**matplotlib version**: 3.3.4

```
cd pydarn/test
pytest test_RTP.py
```

### CODE REVIEW

**Comment by mts299** [comment]
@carleyjmartin ?

**Comment by carleyjmartin** [comment]
Do you want #186 merged in?

**Comment by mts299** [comment]
I think so for the patch release, as long as everything works

File: test/test_Grid.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import bz2
+import datetime as dt
+import matplotlib.pyplot as plt
+import pytest
+import warnings
+
+import pydarn
+
+
+data = pydarn.SuperDARNRead('data/test.grd').read_grid()
```
Comment (carleyjmartin) [inline test/test_Grid.py:24]:
```suggestion
data = pydarn.SuperDARNRead('test/data/test.grd').read_grid()
```

**Review by carleyjmartin — COMMENTED** [review]
Just to round off and check intended usage:
To run it now you must be in the 'pydarn' main directory, either using `pytest --ignore=test/data` or `pytest test/test_Fan.py` for the individual tests.

The only comment is my own fault merging from the other branch, I'll fix it now.

**Review by carleyjmartin — APPROVED** [review]

**Pull request merged**